metricflow/engine/metricflow_engine.py

```diff
@@ -129,7 +129,7 @@ def create_with_random_request_id(  # noqa: D102
         where_constraints: Optional[Sequence[str]] = None,
         order_by_names: Optional[Sequence[str]] = None,
         order_by: Optional[Sequence[OrderByQueryParameter]] = None,
-        sql_optimization_level: SqlQueryOptimizationLevel = SqlQueryOptimizationLevel.O4,
+        sql_optimization_level: SqlQueryOptimizationLevel = SqlQueryOptimizationLevel.default_level(),
```

Comment:
You might need to check if MFS is overriding this param. I know at some point we overrode the `dataflow_plan_optimizations` param at least (to disable predicate pushdown) but not sure now.